### PR TITLE
Feature/te 457 enable drag and drop for macros

### PR DESCRIPTION
**Pull request by georgportwich**

### CODE REVIEW

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropUtils.xtend [hunk 1]
```diff
@@ -117,15 +128,43 @@ class DropUtils {
...
+				}
+				contextIndex++
+			}
+			return null
+		}
 		return EcoreUtil2.getContainerOfType(dropTarget, ComponentTestStepContext)
 	}
 
+	protected def StepContainer getLastStepContext(TclModel tclModel) {
+		if (tclModel.test != null) {
```
Comment (gunther-bachmann) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropUtils.xtend]: if (tclModelTest.test !== null) {
  return tclMode.test.steps.last // <- is null if collection is empty
}

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/UpdateTestModelByDropTarget.xtend [hunk 2]
```diff
@@ -56,46 +57,60 @@ class UpdateTestModelByDropTarget {
...
-		var SpecificationStepImplementation specification = null
-		if (test.steps.size() == 0) {
-			specification = dropUtils.createSpecification
-			toFormatEObject.add(specification)
-			test.steps.add(specification)
-		} else if (test.steps.last.contexts.size() == 0) {
-			specification = test.steps.last
+		var StepContainer specification = null
+		if (tclModel.test != null) {
+			if (tclModel.test.steps.size() == 0) {
```
Comment (gunther-bachmann) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/UpdateTestModelByDropTarget.xtend]: tclModel.test.steps.empty

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/UpdateTestModelByDropTarget.xtend [hunk 3]
```diff
@@ -56,46 +57,60 @@ class UpdateTestModelByDropTarget {
...
+				tclModel.test.steps.add(dropUtils.createSpecification)
+			}
+			specification = EcoreUtil2.getContainerOfType(dropTarget, StepContainer) ?: tclModel.test.steps.last
+		}
+		if (tclModel.macroCollection != null) {
+
+			if (tclModel.macroCollection.macros.size() == 0) {
```
Comment (gunther-bachmann) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/UpdateTestModelByDropTarget.xtend]: empty

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/UpdateTestModelByDropTarget.xtend [hunk 4]
```diff
@@ -56,46 +57,60 @@ class UpdateTestModelByDropTarget {
...
+				tclModel.macroCollection.macros.add(dropUtils.createMacro)
+			}
+			specification = EcoreUtil2.getContainerOfType(dropTarget, StepContainer) ?:
+				tclModel.macroCollection.macros.last
+		}
+
+		if (specification.contexts.size() == 0) {
```
Comment (gunther-bachmann) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/UpdateTestModelByDropTarget.xtend]: empty

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropUtils.xtend [hunk 5]
```diff
@@ -117,15 +128,43 @@ class DropUtils {
 			return (dropTarget.eContainer as SpecificationStepImplementationImpl).getContexts().
 				head as ComponentTestStepContext
 		}
-
+		if (dropTarget instanceof MacroTestStepContextImpl || (dropTarget instanceof TestStepImpl &&
+			dropTarget.eContainer instanceof MacroTestStepContextImpl)) {
+			val macroTestStepContext = EcoreUtil2.getContainerOfType(dropTarget, MacroTestStepContext)
+			val EList<TestStepContext> contexts = EcoreUtil2.getContainerOfType(dropTarget, StepContainer).contexts
+
+			var contextIndex = contexts.indexOf(macroTestStepContext)
```
Comment (gunther-bachmann) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropUtils.xtend]: return contexts.drop(contextIndex).filter(ComponentTestStepContext).head

File: tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropUtils.xtend (hunk 1 above)
Comment (georgportwich) [inline tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropUtils.xtend]: done

File: rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelector.xtend [hunk 6]
```diff
@@ -97,10 +97,15 @@ class TestStepSelector {
 			override done(IJobChangeEvent event) {
 				if (event.job.name.equals("Building workspace")) {
 					logger.info("Building workspace completed. Trigger update TestStepSelector")
-					if (PlatformUI.workbench.activeWorkbenchWindow !== null) {
-						val page = PlatformUI.workbench.activeWorkbenchWindow.activePage
-						broker.post(TestStepSelector.SELECTOR_UPDATE_VIEW, page.activeEditor)
-					}
+					val workbench = PlatformUI.workbench
+
+					logger.debug("active Window: " + workbench)
```
Comment (gunther-bachmann) [inline rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelector.xtend]: Probably better within the syncExec block

Comment (georgportwich) [inline rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelector.xtend]: This was debug-code. Not needed any more. Code was removed

**Comment by gunther-bachmann** [comment]
@Georg: please review my changes

**Pull request merged**